pkg/relay.go

```diff
@@ -526,11 +528,26 @@ func (rs *DefaultRelay) GetValidators(state State) BuilderGetValidatorsResponseE
 	return validators
 }
 
-func (rs *DefaultRelay) verifyBlock(SubmitBlockRequest *types.BuilderSubmitBlockRequest) (bool, error) {
+func (rs *DefaultRelay) verifyBlock(ctx context.Context, SubmitBlockRequest *types.BuilderSubmitBlockRequest, state State) (bool, error) {
+
 	if SubmitBlockRequest == nil {
 		return false, fmt.Errorf("block empty")
 	}
 
+	pubKey := PubKey{SubmitBlockRequest.Message.ProposerPubkey}
+
+	reg, err := state.Datastore().GetRegistration(ctx, pubKey)
```

Comment:
This is quite inefficient in the current setting, since it requires a DB io, in every block submission. Instead, we should keep this information in memory every epoch / slot. But let's wait for https://github.com/blocknative/dreamboat/pull/37 before implementing this.

Reply:
Great idea!